src/cargo/command_bridge.rs

```diff
@@ -0,0 +1,141 @@
+//! This struct is a simple bridge that we can inspect for test purpose
+
+use std::collections::HashMap;
+use std::convert::{AsRef, Into};
+use std::default::Default;
+use std::ffi::{OsStr, OsString};
+use std::process::{Command, Stdio};
+
+/// This temporary struct is used to inspect the generated command before calling the process::Command
+/// builder, as we can't inspect its structure and make assertions on it
+#[derive(Debug, Default)]
+pub struct CommandBridge {
+    pub name: String,
+    pub args: Vec<OsString>,
+    pub env: HashMap<OsString, OsString>,
+    pub stdout: Option<Stdio>,
+    pub stderr: Option<Stdio>,
+}
+
+impl CommandBridge {
+    pub fn new(name: &str) -> Self {
+        CommandBridge {
+            name: String::from(name),
+            .. Default::default()
+        }
+    }
+
+    pub fn arg<S: AsRef<OsStr>>(&mut self, arg: S) -> &mut Self {
+        let mut os_string = OsString::new();
+        os_string.push(arg);
+        self.args.push(os_string);
+        self
+    }
+
+    pub fn env<S: AsRef<OsStr>>(&mut self, key: S, value: S) -> &mut Self {
+        let mut key_str = OsString::new();
+        let mut value_str = OsString::new();
+
+        key_str.push(key.as_ref());
+        value_str.push(value.as_ref());
+
+        self.env.insert(key_str, value_str);
+        
+        self
+    }
+
+    pub fn stderr<S: Into<Stdio>>(&mut self, stderr: S) -> &mut Self {
+        self.stderr = Some(stderr.into());
+        self
+    }
+
+    pub fn stdout<S: Into<Stdio>>(&mut self, stdout: S) -> &mut Self {
+        self.stdout = Some(stdout.into());
+        self
+    }
+
+    pub fn to_command(self) -> Command {
+        let mut command = Command::new(self.name);
```

Comment:
I don't think `name` is the best choice to describe this. I think `command` would be better but that looks kind of weird here.

Reply:
good point. `command_name` it is

Reply:
Awesome thanks! :D